### DISCUSSION TITLE
Funktionierende Lösung für Tasmota und Siemens IM350 Wienernetze IR Leser Hichi IR TTL Olimex PoE ISO

### DISCUSSION THREAD

**Discussion by dpunzeng** [arendst/Tasmota#20755] [Q&A]
Hello All, hallo Alle! 
**Funktionierende Lösung für Tasmota und Siemens IM350 Wienernetze IR Leser Hichi IR TTL Olimex PoE ISO** 

Ich habe jetzt einen halben Tag herumgeschissen, um den Wienernetze Siemens IM350 mit Tasmota auf Olimex PoE Iso auszulesen.
Folgende Punkte muss man wissen:
1) Tasmota selbst kompilieren mit https://tasmota.github.io/docs/Compile-your-build/
2) Ports **GPIO1 auf TX** und **GPIO3 auf RX**
3) Die Hardware testen, zuerst **mit hartem Loopback**, dann mit **Spiegel am IR**. Debug aufdrehen mit "sensor35 d1" in der Konsole
4) Erst wenn das geht, den IR Kopf anbringen.
5) Der Schlüssel von **Wienernetze Web ändert** sich offensichtlich nochmal, nachdem man das aktiviert hat (das hat mir 3h gekostet)
5) Der **Offset der echten Payload** muss im Script eingestellt werden. Dazu die entschlüsselten Pakete (56Bytes) auslesen und die ersten 3 Byte im Auslesescript eintragen: Bsp: 
```
15:52:42.789 SML: decrypted block: 56 bytes
15:52:42.792 :>**02 09 09** 0c 07 e8 02....
```
Die ersten drei Byte sind "020909"
Diese müssen im Script beim Parser eingetragen werden wie folgt:
`1,**020909**x19UUuu@1,year,,year,0`

Nur genau dann geht es und man muss schon halbwegs Hacker-Energien haben um das alles zu sammeln.
Hier noch das Wienernetze Script OHNE meinem Key natürlich, der ist ersetzt durch "KEY"
>>

```
>D
>B
=>sensor53 r
>M 1
+1,3,r,0,9600,Home
1,=so3,128
1,=so4,KEY<----------------------------------------hier
1,020909x19UUuu@1,year,,year,0
1,020909x21ss@1,month,,month,0
1,020909x22ss@1,day,,day,0
1,020909x24ss@1,hh,,hh,0
1,020909x25ss@1,mm,,mm,0
1,020909x26ss@1,ss,,ss,0
1,020909x32UUuuUUuu@1000,+A,kWh,+A,3
1,020909x37UUuuUUuu@1000,-A,kWh,-A,3
1,020909x42UUuuUUuu@1000,+R,varh,+R,3
1,020909x47UUuuUUuu@1000,-R,varh,-R,3
1,020909x52UUuuUUuu@1,+P,W,+P,3
1,020909x57UUuuUUuu@1,-P,W,-P,3
1,020909x62UUuuUUuu@1,+Q,var,+Q,3
1,020909x67UUuuUUuu@1,-Q,var,-Q,3
#
```
Edit:
Der datastream hängt offensichtlich auch noch von irgendwelchen Faktoren ab. Der oben gezeigte, bringt nicht das richtige Mapping bei mir. Ist mir erst im Nachgang aufgefallen. Diese beiden Tags bringen Verbrauch (KWh) und Leistung (W)
Nur durch probieren gefunden :(
```
1,020909x14UUuuUUuu@1000,Zählerstand,kWh,Zaehlerstand,2
1,020909x34UUuuUUuu@1000,Bezug aktuell,kwh,Bezug,3]
```

**Comment by my-machines** [comment]
Vielen Dank für Dein Script! Hast du vielleicht auch herausgefunden wie man die momentane Einspeiseleistung auslesen kann?

**Reply by Noschvie** [reply]
Debug Output erstellen und in den SML Decoder einwerfen und auswerten...

**Reply by my-machines** [reply]
Danke für den Tipp, mein SM liefert immer nur 5 Zeilen und die kann der SML Decoder nicht übersetzen :/

**Reply by my-machines** [reply]
in einem anderen Forum gefunden: 

```
1,020909x19UUuuUUuu@1000,Einspeisung Gesamt,kWh,Einspeisung_ges,2 
1,020909x39UUuuUUuu@1000,Einspeisung aktuell ,kW,Einspeisung_akt,3 
1,020909x29UUuuUUuu@1000,BL Einspeisung,kvarh,BL_Einspeisung,3
```

**Reply by gimx** [reply]
https://www.107er.net/hitech/2024/09/howto-smartmeter-mit-tasmota-in-home-assistant-einbinden/ funktioniert bei mir

**Comment by hs-flo** [comment]
Darf ich um die gesamten Daten eines  Zyklus deines Zählers bitten? bzw. um das fertige image ebenso (bzw die Einstellungen, habe Tasmota noch nie kompilieren müssen).
Ich scheitere am NetzOÖ ebenso an der Entschlüsselung. Solabd ich den Key aktiviere, bekomme ich keine Ausgeben mehr. DAvor pro Sekunde je:
14:24:24.993 : 68 5f 5f 68 53 f0 5b 00 00 00 00 2d 4c 01 0e ae 00 50 05 f8 32 86 dd f8 94 40 a2 ed 92 47 3a c7 
14:24:25.030 : 13 cb c8 07 3c 15 1b a1 70 5b c0 52 d2 30 da 8a c8 8f 7f 0e 70 89 5c 6f 28 67 e5 d0 d3 4a 90 05 
14:24:25.066 : 59 40 c7 04 cf 67 78 07 ab 3d 4e 92 4c 1d 39 8b 46 48 48 e4 45 66 44 70 82 06 eb fb fc 9a 4e bf 
14:24:25.112 : 05 8d 7d cd 16 

Passt mit der Anleitung des Betreibers genau zusammen, Payload ist verschlüsselt. Nun habe ich schon mehrfache Versuche mit Offsets zur Payload bzw. Teilen des Frames, bekomme aber am Remote Syslog bei aktiviertem 1,=s04 "unspecified error while reading data". Keinerlei Ausgabe in der Tasmota Console.
Habe bis zu dem Punkt die Anleitung der NetzOÖ verfolgt: https://www.netzooe.at/2021-11-05-spezifikation-ks-amis.pdf?ch=ttzaxj8g&:hp=3;2;de
Ich würde mich über Tips und Hilfe sehr freuen. 
Danke, Florian
